src/main/java/org/folio/service/finance/transaction/summary/OrderTransactionSummariesService.java

```diff
@@ -29,9 +28,20 @@ public Future<Void> updateTransactionSummary(String orderId, int number, Request
       return updateTransactionSummary(summary, requestContext);
     }
 
-    public Future<Void> createOrUpdateOrderTransactionSummary(EncumbrancesProcessingHolder holder, RequestContext requestContext) {
-      Promise<Void> promise = Promise.promise();
+  public Future<Void> updateOrCreateTransactionSummary(String orderId, int number, RequestContext requestContext) {
+    return getTransactionSummary(orderId, requestContext)
+      .recover(t -> null)
```

Comment:
I suggest to focus only on 404 response code and other case throw exception

Reply:
I added a check. I hope we will not need this hack eventually, after we fix the finance API.